src/Transition/CreateNewIDFUsingRulesV9_6_0.f90

```diff
@@ -397,7 +397,9 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
                   CALL GetNewObjectDefInIDD(ObjectName,NwNumArgs,NwAorN,NwReqFld,NwObjMinFlds,NwFldNames,NwFldDefaults,NwFldUnits)
                   nodiff=.false.
                   OutArgs(1:3)=InArgs(1:3)
-                  CurArgs = CurArgs - 1
+                  IF (CurArgs == 4) THEN
+                      CurArgs = 3
```

Comment:
That looks good! I tested again on both a 4-line case and a 3-line case, it works fine on both cases.